src/main/java/interface_adapter/student_logged_in/student_home/show_clubs/StudentShowClubsPresenter.java

```diff
@@ -23,9 +23,10 @@ public StudentShowClubsPresenter(StudentHomeViewModel studentHomeViewModel, View
     @Override
     public void preparePostContent(StudentShowClubsOutputData studentShowClubsOutputData) {
         final StudentHomeState state = studentHomeViewModel.getState();
-        final ArrayList<String> clubNames = (ArrayList<String>) studentShowClubsOutputData.getClubs();
+        // TODO make it so output data ONLY has a list of the names, not an entire list of maps with club data
+        // final ArrayList<String> clubNames = (ArrayList<String>) studentShowClubsOutputData.getClubs();
```

Comment:
I agree with this.

Reply:
Currently, the output data has an arraylist containing maps, where each map has the data for a club. You cannot cast an arraylist of maps into an arraylist of strings. The extra data in the maps is also not necessary, since the presenter only ever uses the names of the clubs.